test(e2e,playwright): profile.tsx

leonfeng opened the pull request:
Checklist:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x]. It will ensure that our team takes your pull request seriously. -->

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.-->

Ref #51705

<!-- Feel free to add any additional description of changes below this line -->
I'm not entirely sure what I'm doing. So please don't hesitate to request any changes.

Thank you for reviewing my PR :)

leonfeng commented:
camperbot cant code

naomi-lgbt commented on client/src/pages/__snapshots__/email-sign-up.test.js.snap:
```diff
@@ -162,7 +162,7 @@ exports[`<EmailSignUp /> Non-Authenticated user "not accepted terms and conditio
       <a
         className="big-cta-btn btn btn-lg btn-primary btn-block"
-        href="http://localhost:3000/signin"
+        href="https://3000-leonfeng-freecodecamp-z4lp4gb2uwf.ws-us105.gitpod.io/signin"
```
I don't believe this should be changed.

leonfeng replied:
Thanks for reviewing my PR, Naomi. I'm currently studying Gitpod documentation. Will @ mention you when I think everything is fixed.

naomi-lgbt commented on e2e/profile.spec.ts:
```diff
@@ -0,0 +1,102 @@
+
+let page: Page;
+
+test.beforeAll(async ({ browser }) => {
+  page = await browser.newPage();
+  await page.goto('/certifieduser');
+  await page.getByRole('button', { name: 'Preview custom 404 page' }).click();
+});
+
+test.afterAll(async () => {
+  await page.close();
+});
```
```suggestion
test.beforeEach(async ({ page }) => {
  await page.goto('/certifieduser');
  await page.getByRole('button', { name: 'Preview custom 404 page' }).click();
});

test.afterEach(async ({ page }) => {
  await page.close();
});
```

Ideally we should run the tests in isolation.

naomi-lgbt commented on e2e/profile.spec.ts:
```diff
@@ -0,0 +1,102 @@
+import { test, expect, type Page } from '@playwright/test';
```
```suggestion
import { test, expect } from '@playwright/test';
```

Remove the import because...

naomi-lgbt commented on e2e/profile.spec.ts:
```diff
@@ -0,0 +1,102 @@
+
+test.use({ storageState: 'playwright/.auth/certified-user.json' });
+
+let page: Page;
+
```
```suggestion
```

We don't need the global `page`, because...

naomi-lgbt commented on e2e/profile.spec.ts:
```diff
@@ -0,0 +1,102 @@
+});
+
+test.describe('Profile component', () => {
+  test('renders the camper profile correctly', async () => {
```
```suggestion
  test('renders the camper profile correctly', async ({ page }) => {
```

We can get `page` directly from the function args :3

naomi-lgbt commented on e2e/profile.spec.ts:
```diff
@@ -0,0 +1,102 @@
+    await expect(page.getByText('2019')).toBeVisible();
+  });
+
+  test('renders total points correctly', async () => {
```
```suggestion
  test('renders total points correctly', async ({ page }) => {
```

naomi-lgbt commented on e2e/profile.spec.ts:
```diff
@@ -0,0 +1,102 @@
+    await expect(page.getByText('Number of points: 1')).toBeVisible();
+  });
+
+  test('renders the heat map correctly', async () => {
```
```suggestion
  test('renders the heat map correctly', async ({ page }) => {
```

naomi-lgbt commented on e2e/profile.spec.ts:
```diff
@@ -0,0 +1,102 @@
+    await expect(page.locator('div.streak-container')).toBeVisible();
+  });
+
+  test('displays certifications correctly', async () => {
```
```suggestion
  test('displays certifications correctly', async ({ page }) => {
```

naomi-lgbt commented on e2e/profile.spec.ts:
```diff
@@ -0,0 +1,102 @@
+    ).toEqual(certs.length + legacyCerts.length);
+  });
+
+  test('should not show portfolio when empty', async () => {
```
```suggestion
  test('should not show portfolio when empty', async ({ page }) => {
```

naomi-lgbt commented on e2e/profile.spec.ts:
```diff
@@ -0,0 +1,102 @@
+    ).toEqual(0);
+  });
+
+  test('displays the timeline correctly', async () => {
```
```suggestion
  test('displays the timeline correctly', async ({ page }) => {
```

leonfeng committed:
test(e2e,playwright): profile.tsx

leonfeng commented:
Thank you for the informative review, Huyen! I've implemented all the requested changes. I've also replaced most of the other locators with get by role or text.

However, some tests are still failing. I'll investigate tomorrow. Cheers~

huyenltnguyen committed:
remove afterEach block

huyenltnguyen committed:
ensure the button exists before clicking

huyenltnguyen committed:
comment out the preview button click

leonfeng committed:
Debug tests

leonfeng commented on e2e/profile.spec.ts:
```diff
@@ -0,0 +1,159 @@
+  test('renders the camper profile correctly', async ({ page }) => {
+    await expect(
+      page
+        .getByText('ProfileDefault AvatarAn avatar coding with a laptop')
```
The `DefaultAvatar` component has no ARIA role. This is the locator suggested by Playwright's VS Code extension.

huyenltnguyen committed:
refactor: remove test label and update query methods

huyenltnguyen reviewed (approved):
I went ahead and made some small updates myself to shorten the review loop: https://github.com/freeCodeCamp/freeCodeCamp/commit/c39c94d614f721dd208401b923fb18309b096bf6.

Thank you for your contribution @leonfeng! And also, thanks for bearing with us 😄
